### DISCUSSION TITLE
Is there a way to make a more desktop profiles for specific applications?

### DISCUSSION THREAD

**Discussion by nikolayminchevdev** [Valkirie/HandheldCompanion#812] [Q&A] [answered]
Hello,

Currently I am struggling to make profile for games that do not have controller support like StarCraft 2, WarCraft and older games. They use the default desktop layout and they do not have the option for personalized one. 

If I make a profile for StarCraft 2 and change the button mappings in the controller layout they are not applied because the game only uses a desktop layout. If I want specific layouts for such games I have to manually remap buttons every time I want to start such game/application.

Is there a way to achieve that?

**Comment by Valkirie** [comment]
You need to create a profile for Starcraft 2. Either via the main UI or quicktools.
![image](https://github.com/Valkirie/HandheldCompanion/assets/934757/141624e0-346f-43b9-8b2d-e6a990a5433b)

Then you need to turn on Controller layout
![image](https://github.com/Valkirie/HandheldCompanion/assets/934757/6964af6a-43c5-441e-a480-3fb468882a83)

Here you can change buttons and actions specifically for Starcraft. You can use one of our template from the list or create your own.
![image](https://github.com/Valkirie/HandheldCompanion/assets/934757/ddaa9f32-2873-4e7d-9f8a-015a8965cd99)

**Reply by nikolayminchevdev** [reply]
Actually, it worked but I had to minimize the app and turn off the desktop layout, otherwise it is prioritized and the game specific one is not applied.

A workaround I found is to make the default controller profile a desktop one and just turn off the one from the main Controller menu (non-game controller layouts).

**Comment by Valkirie** [comment, marked as answer]
- Do you have Desktop mode enabled on HC ? If yes, turn if off.
- Make sure Steam is not messing with inputs otherwise (steam desktop chord is disabled or empty).

**Reply by nikolayminchevdev** [reply]
Yes, I had to turn off the "Non-game controller layout".

A workaround I found is to make the default controller profile a desktop one (so after exiting the game I can browse without issue in desktop mode) and just turn off the one from the main Controller menu (non-game controller layouts).
